providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/EventStreamObserver.java

```diff
@@ -0,0 +1,101 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import java.util.Map;
+import io.grpc.stub.StreamObserver;
+import dev.openfeature.flagd.grpc.Schema.EventStreamResponse;
+import dev.openfeature.sdk.ProviderEvaluation;
+import com.google.protobuf.Value;
+
+/**
+ * EventStreamObserver handles events emitted by flagd.
+ */
+public class EventStreamObserver implements StreamObserver<EventStreamResponse> {
+    private EventStreamCallback callback;
+    private Boolean cacheEnabled;
+    private Map<String, ProviderEvaluation<Boolean>> booleanCache;
+    private Map<String, ProviderEvaluation<String>> stringCache;
+    private Map<String, ProviderEvaluation<Double>> doubleCache;
+    private Map<String, ProviderEvaluation<Integer>> integerCache;
+    private Map<String, ProviderEvaluation<dev.openfeature.sdk.Value>> objectCache;
```

Comment:
Is there a reason to maintain 5 caches besides the need to support multiple types?

I think we could have one cache of type `Object` and Handle everything that way, casting as needed since we know the type can't change while in the cache.

Reply:
I considered doing the same thing while working on the Go implementation (storing as `interface{}`) but decided against it to avoid the overhead of type casting in the code and at runtime. Would be good to get some other opinions on this as there are definitely pros/cons to each approach
@james-milligan @Kavindu-Dodan @beeme1mr

Reply:
And regarding type casting, multiple maps do avoid the requirement of type casting. But, that brings other concerns like the cache size and maintenance of them. 

If we opt to use `Value` from sdk, we can abstract out the casting issue to the Value class (it does use downcasting from Object). And later, we could improve `Value` class internals to avoid casting if we see a big issue with casting.

Reply:
Thanks for the suggestions @Kavindu-Dodan , I've implemented as such :)